### DISCUSSION TITLE
Live Wallpapers and ScreenSaver without external apps

### DISCUSSION THREAD

**Discussion by ZicPL** [pop-os/cosmic-epoch#1408] [Polls]
Hi all,
There is a request to implement the title features in this issue: [#605](https://github.com/pop-os/cosmic-epoch/issues/605)
I'm an optimist and I know that my ideas are sometimes impossible or too expensive to implement. With that in mind I didn't want to focus only on the discussion, but to find out if I'm not the only one who thinks so 😇

A survey seems like a fair way to find out without focusing only on my opinion.

Best regards
ps. I've high hopes for the new DE

**Comment by pL-uTo** [comment]
This is very difficult to implement. You have to take into account the small experience of the team creating Cosmic, implementation in Wayland and Gnome, KDE with huge feedback didn't succeed in achieving this goal..

**Reply by ZicPL** [reply]
Does this also confirm @jackpot51 
I think the Dev Team has enough experience to implement this.

**Comment by DGMurdockIII** [comment]
Please yes

**Comment by meraklimaymun** [comment]
+1 on this.

This is not even available on Windows natively! I was running Wallpaper Engine on Steam/Windows for this feature. When I switched to Linux, that app didn't work. I checked but there are no equivalent in Linux. If Cosmic provided this natively, that would put it above all other desktop environments.

But this is a huge undertaking. There has to be a custom scripting language for users to make their own 2D, 3D animations for live wallpapers. There needs to be a store for users to browse, download, upload, favourite their designs. There needs to be moderation for NSFW content.

**Reply by ZicPL** [reply]
There're more benefits than disadvantages. I wouldn't like to list everything here because it would be a very long post.

In my case, I'm looking for a solution that will allow me to protect the OLED screen. I've a video with RGBW colors and in Windows I've a screensaver that shows me this video every minute as well as a live wallpaper that prevents the static image from being displayed. Thanks to this, I protect the screen from "Screen Burn-In". There aren't such apps in Linux. If I do find a solution in Linux, it takes a lot of resources and energy (reduces system performance). The most efficient solution seems to me to be built into DE (desktop environment).

Cosmic is still being developed and it seems to me that this is the best time to implement such a feature - later it may cost more work.

**Comment by DGMurdockIII** [comment]
yes i know this sint gnome but this might be of use  Old-school screensaver functionality for GNOME/Wayland desktops  https://gist.github.com/vdbsh/a9f0723708a4393d42a0d768d831c4df

**Reply by ZicPL** [reply]
Screensaver, live wallpaper, and more similar solutions -> You can try to implement it yourself (even put together different solutions from ready-made blocks - other apps). I'm more than certain that it won't be as efficient as the native DE feature.

**Reply by n3tea** [reply]
> native DE feature

These apps don't work well on any DE (using Wyland). Support for feature by DE under Wyland would be a great achievement. I believe it will succeed.

**Comment by Net0S** [comment]
I wonder how devs have mastered Wayland 🤔 

For me, this would be the most interesting feature in DE in many years. I see a lot of interest in this feature and I wonder if anyone will pay attention to the community's request.

**Reply by DGMurdockIII** [reply]
Hope so

**Comment by UsrD3f** [comment]
I'd like to have this feature. Unfortunately, I don't think such a small team can migrate the library responsible for this to RUST. Especially since the original Wyland library responsible for screensavers (background animations) doesn't work as well as we'd like. This is more of a task for larger teams, like those from Gnome or Plasma.

**Reply by n3tea** [reply]
true
